src/events/mouse.js

```diff
@@ -311,20 +311,26 @@ p5.prototype.isMousePressed = false; // both are supported
 p5.prototype._updateNextMouseCoords = function(e) {
   var x = this.mouseX;
   var y = this.mouseY;
+  var winX = this.winMouseX;
+  var winY = this.winMouseY;
   if(e.type === 'touchstart' ||
      e.type === 'touchmove' ||
      e.type === 'touchend' || e.touches) {
     x = this.touchX;
     y = this.touchY;
+    winX = this.winTouchX;
+    winY = this.winTouchY;
   } else if(this._curElement !== null) {
     var mousePos = getMousePos(this._curElement.elt, e);
     x = mousePos.x;
     y = mousePos.y;
+    winX = mousePos.winX;
+    winY = mousePos.winY;
   }
   this._setProperty('mouseX', x);
   this._setProperty('mouseY', y);
-  this._setProperty('winMouseX', e.pageX);
-  this._setProperty('winMouseY', e.pageY);
+  this._setProperty('winMouseX', winX);
+  this._setProperty('winMouseY', winY);
```

Comment:
Yeah, I think if we just updated these two lines to use clientX/Y, it'd work fine for the mouse position.

The issue I ran into was that touch events don't have clientX/Y, so it has to be grabbed from the touches array (lines 196-197 in touch.js). Then everything else is just to pass that to winMouseX/Y. You'll notice I mimicked the way that touchX/Y is passed to mouseX/Y.

Looking at it now, we could just let the touch events populate the mouse properties directly. Then we could drop touchX/Y, as it's essentially a duplicate of mouseX/Y, right?